src/ray/core_worker/transport/actor_task_submitter.cc

```diff
@@ -282,13 +282,22 @@ void ActorTaskSubmitter::FailInflightTasks(
   }
 }
 
+size_t ActorTaskSubmitter::NumInflightTasks(const ActorID &actor_id) const {
```

Comment:
is it common to add a function only for testing?

Reply:
No this is not best practice, we should avoid doing this unless necessary

Reply:
After reading the code I think we can delete this entirely and the tests wouldn't lose any value

Reply:
I think it's good to check the current number of inflight tasks to make sure we don't finish/fail the wrong task. 

Since it's read only function, I think it's better than marking the test as a friend and have full access to the private fields?